SIPS/sip-29.md

```diff
@@ -35,85 +35,147 @@ in uppercase in this document are to be interpreted as described in [RFC
+//         },
+//         {
+//             name: 'wei',
+//             ticker: 'wei',
+//             decimals: 0
+//         }
+//    ]
+// }
+
+// Represents a asset unit.
+type AssetUnit = {
+    // Human-friendly name of the asset unit.
     name: string;
 
-    // Ticker of the token unit.
+    // Ticker of the asset unit.
     ticker: string;
 
-    // Number of decimals of the token unit.
+    // Number of decimals of the asset unit.
     decimals: number;
 };
 
-// Token description.
-type TokenDescription = {
-    // Human-friendly name of the token.
+// Asset description.
+type AssetDescription = {
+    // The CAIP-19 ID of the asset.
+    id: Caip19AssetType;
+
+    // Human-friendly name of the asset.
     name: string;
 
-    // Ticker of the token.
+    // Ticker of the asset.
     ticker: string;
 
-    // Whether the token is native to the chain.
+    // Whether the asset is native to the chain.
     isNative: boolean;
 
-    // Base64 representation of the token icon.
+    // Base64 representation of the asset icon.
     iconBase64: string;
 
-    // List of token units.
-    units: TokenUnit[];
+    // List of asset units.
+    units: AssetUnit[];
 };
 
-// Returns the description of a non-fungible token. This description can then
-// be used by the client to display relevant information about the token.
-//
-// @example
-// ```typescript
-// const tokenDescription = await getTokenDescription('eip155:1/slip44:60');
-//
-// // Returns:
-// // {
-// //     name: 'Ether',
-// //     ticker: 'ETH',
-// //     isNative: true,
-// //     iconBase64: 'data:image/png;base64,...',
-// //     units: [
-// //         {
-// //             name: 'Ether',
-// //             ticker: 'ETH',
-// //             decimals: 18
-// //         },
-// //         {
-// //             name: 'Gwei',
-// //             ticker: 'Gwei',
-// //             decimals: 9
-// //         },
-// //         {
-// //             name: 'wei',
-// //             ticker: 'wei',
-// //             decimals: 0
-// //         }
-// //     ]
-// // }
-// ```
-function getTokenDescription(token: Caip19AssetType): TokenDescription;
+
+type OnAssetDescriptionReturn = {
+    description: Record<Caip19AssetType, AssetDescription>;
+};
 ```
 
-#### Get Token Conversion Rate
+#### Get Asset Conversion Rate
+
+```typescript
+import { OnAssetConversionHandler } from "@metamask/snaps-sdk";
+
+export const onAssetConversion: OnAssetConversionHandler = async ({
+  from,
+  to
+}) => {
+  const conversionRate = /* Get conversion rate */;
+  return { conversionRate };
+};
+```
+The type for an `onAssetDescription` handler function’s arguments is:
 
 ```typescript
-type TokenConversionRate = {
-    // The rate of conversion from the source token to the target token. It
-    // means that 1 unit of the `from` token should be converted to this amount
-    // of the `to` token.
+interface OnAssetConversionArgs {
+    from: Caip19AssetType[];
+    to: Caip19AssetType;
```

Comment:
I'm wondering if we should also support multiple `to`

Reply:
I think it makes more sense to have `{from, to}[]`